client/my-sites/plans/plan-overview/plan-status/index.jsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import ProgressBar from 'components/progress-bar';
+import { isPremium, isBusiness } from 'lib/products-values';
+
+const PlanStatus = React.createClass( {
+	propTypes: {
+		plan: React.PropTypes.object.isRequired
+	},
+
+	getDaysUntilExpiry() {
+		const { userFacingExpiryMoment } = this.props.plan;
+
+		return userFacingExpiryMoment.diff( this.moment(), 'days' );
+	},
+
+	renderProgressBar() {
+		const { subscribedMoment, userFacingExpiryMoment } = this.props.plan,
+			// we strip the hour/minute/second/millisecond data here from `subscribed_date` to match `expiry`
+			trialPeriodInDays = userFacingExpiryMoment.diff( subscribedMoment, 'days' ),
+			timeUntilExpiryInDays = this.getDaysUntilExpiry(),
+			progress = Math.max( 0.5, trialPeriodInDays - timeUntilExpiryInDays );
+
+		return (
+			<ProgressBar
+				value={ progress }
+				total={ trialPeriodInDays } />
+		);
+	},
+
+	renderDaysRemaining() {
+		if ( this.getDaysUntilExpiry() > 1 ) {
+			return this.translate( '%(daysUntilExpiry)s days remaining', {
+				args: { daysUntilExpiry: this.getDaysUntilExpiry() },
+				context: 'The amount of time until the trial plan expires, e.g. "5 days remaining"'
+			} );
+		}
+
+		return this.translate( '1 day remaining' );
```

Comment:
We should probably use the `count` attribute here in order to better handle [pluralization](https://github.com/Automattic/calypso-pre-oss/tree/master/client/lib/mixins/i18n#pluralization).